### PR TITLE
Minor refactors

### PR DESCRIPTION
**Pull request by amrsoll**
Some refactoring to make the code more "python'esque" and hopefully more readable. 

- Function docstrings
- Move ``def function(self, ... )`` inside the appropriate classes
- Break down the data reading from the icons
- Remove some unused functions / unnecessary lines
- move most imports to the start of the file
- A little bit of function typing
- Remove global function `tag_update` -> Classfunction. Less confusing because of blender function with same name

Note: I assumed that `tag_update` does not refer to the `bpy` `tag_update` function. Here I make an explicit reference to `SlvsGenericEntity.tag_update()`

I have done minor testing to see if I didn't break anything major

N.B. I would recommend using a develop branch for people to create PRs against :) just for sanity check before commiting multiple PRs into master

### CODE REVIEW

File: operators.py [hunk 1]
```diff
@@ -743,7 +743,6 @@ def register_properties(cls):
         for a in annotations.keys():
             if hasattr(cls, a):
                 raise NameError("Cannot register implicit pointer properties, class {} already has attribute of name {}".format(cls, a))
-        setattr(cls, "__annotations__", annotations)
```
Comment (amrsoll) [inline operators.py:746]: `annotations` was never modified

File: testing/utils.py [hunk 2]
```diff
@@ -45,8 +45,6 @@ class Sketch2dTestCase(BgsTestCase):
     def setUpClass(cls):
         super().setUpClass()
 
-        from mathutils import Quaternion
```
Comment (amrsoll) [inline testing/utils.py:48]: unused

File: class_defines.py [hunk 3]
```diff
@@ -660,12 +662,9 @@ def normal(self):
 ]
 
 
-def hide_sketch(self, context):
-    if self.convert_type != "NONE":
-        self.visible = False
```
Comment (amrsoll) [inline class_defines.py:665]: unused

**Comment by hlorus** [comment]
LGTM, thanks for the PR!

**Pull request merged**